parser/grammar.peg

```diff
@@ -196,17 +196,20 @@ Field ← id:IntConstant _ ':' _ req:FieldReq? _ typ:FieldType _ name:Identifier
 		Type     : typ.(*Type),
 		Annotations: toAnnotations(annotations),
 	}
-	if req != nil && !req.(bool) {
-		f.Optional = true
-	}
+    switch requiredness, _ := req.(string); requiredness {
+    case "required":
+        f.Required = true
+    case "optional":
+        f.Optional = true
+    }
 	if def != nil {
 		f.Default = def.([]interface{})[2]
 	}
 	return f, nil
 }
 
 FieldReq ← ("required" / "optional") {
-	return !bytes.Equal(c.text, []byte("optional")), nil
+	return string(c.text), nil
```

Comment:
Not quite clear what's going on here - I'm guessing the signature's changing from bool to string as there are now three states to reflect? I'm not clear on what this FieldReq ← ("required" / "optional") bizniz is